q2_vsearch/tests/test_cluster_features.py

```diff
@@ -7,63 +7,71 @@
 # ----------------------------------------------------------------------------
 
 import os
+import sqlite3
+import tempfile
+
 import skbio
 import biom
 import numpy as np
-import tempfile
 
+from qiime2 import Artifact
 from qiime2.plugin.testing import TestPluginBase
 from qiime2.util import redirected_stdio
-from q2_types.feature_data import DNAFASTAFormat
+from q2_types.feature_data import DNAFASTAFormat, DNAIterator
 
 from q2_vsearch._cluster_features import (cluster_features_de_novo,
                                           cluster_features_closed_reference,
-                                          _fasta_with_sizes)
+                                          _fasta_with_sizes,
+                                          _fasta_from_sqlite)
+
+
+def _read_seqs(seqs):
+    if isinstance(seqs, DNAFASTAFormat):
+        read_seqs = skbio.io.read(str(seqs), constructor=skbio.DNA,
+                                  format='fasta')
+    elif isinstance(seqs, str):
+        read_seqs = skbio.io.read(seqs, constructor=skbio.DNA, format='fasta')
+    else:
+        read_seqs = seqs.view(DNAIterator)
+    return list(read_seqs)
 
 
 class ClusterFeaturesDenovoTests(TestPluginBase):
 
     package = 'q2_vsearch.tests'
 
-    def test_no_clustering(self):
+    def setUp(self):
+        super().setUp()
         input_sequences_fp = self.get_data_path('dna-sequences-1.fasta')
-        input_sequences = DNAFASTAFormat(input_sequences_fp, mode='r')
-        input_table = biom.Table(np.array([[100, 101, 103],
-                                           [1, 1, 2],
-                                           [4, 5, 6],
-                                           [7, 8, 9]]),
-                                 ['feature1', 'feature2', 'feature3',
-                                  'feature4'],
-                                 ['sample1', 'sample2', 'sample3'])
+        self.input_sequences = DNAFASTAFormat(input_sequences_fp, mode='r')
+        self.input_table = biom.Table(np.array([[100, 101, 103],
+                                                [1, 1, 2],
+                                                [4, 5, 6],
+                                                [7, 8, 9]]),
+                                      ['feature1', 'feature2', 'feature3',
+                                       'feature4'],
+                                      ['sample1', 'sample2', 'sample3'])
+        self.exp_seqs = _read_seqs(self.input_sequences)
```

Comment:
I think we should change the name `self.exp_seqs` - maybe `self.input_sequences_list` (which also isn't ideal, but we have the input sequences as two different types here).